docset/winserver2022-ps/netsecurity/Rename-NetFirewallHyperVRule.md

```diff
@@ -0,0 +1,345 @@
+---
+description: Use this topic to help manage Windows and Windows Server technologies with Windows PowerShell.
+external help file: NetFirewallHyperVRule.cmdletDefinition.cdxml-help.xml
+Module Name: NetSecurity
+ms.date: 8/25/2023
+online version: https://docs.microsoft.com/powershell/module/netsecurity/rename-netfirewallhypervrule?view=windowsserver2022-ps&wt.mc_id=ps-gethelp
+schema: 2.0.0
+title: Rename-NetFirewallHyperVRule
+---
+
+# Rename-NetFirewallHyperVRule
+
+## SYNOPSIS
+Renames a single Hyper-V firewall rule.
+
+## SYNTAX
+
+### GetAll (Default)
+```
+Rename-NetFirewallHyperVRule -NewName <string> [-All] [-PolicyStore <string>] [-CimSession <CimSession[]>] [-ThrottleLimit <int>] [-AsJob] [-PassThru] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### ByName
+```
+Rename-NetFirewallHyperVRule [-Name] <string[]> -NewName <string> [-PolicyStore <string>] [-CimSession <CimSession[]>] [-ThrottleLimit <int>] [-AsJob] [-PassThru] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### ByDisplayName
+```
+Rename-NetFirewallHyperVRule -DisplayName <string[]> -NewName <string> [-PolicyStore <string>] [-CimSession <CimSession[]>] [-ThrottleLimit <int>] [-AsJob] [-PassThru] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### ByQuery
+```
+Rename-NetFirewallHyperVRule -NewName <string> [-Direction {Inbound | Outbound}] [-VMCreatorId <string[]>] [-Protocol <string[]>] [-Action {NotConfigured | Allow | Block}] [-Enabled {True | False}] [-EnforcementStatus {Unknown | OK | PartiallyEnforced | NoApplicablePorts | ParsingError | Error}] [-PolicyStoreSourceType {None | Local | MDM}] [-PolicyStore <string>] [-CimSession <CimSession[]>] [-ThrottleLimit <int>] [-AsJob] [-PassThru] [-WhatIf] [-Confirm]  [<CommonParameters>]
+```
+
+### InputObject (cdxml)
+```
+Rename-NetFirewallHyperVRule -InputObject <CimInstance#MSFT_NetFirewallHyperVRule[]> -NewName <string> [-CimSession <CimSession[]>] [-ThrottleLimit <int>] [-AsJob] [-PassThru] [-WhatIf] [-Confirm]  [<CommonParameters>]
+```
+
+## DESCRIPTION
+The **Rename-NetFirewallHyperVRule** cmdlet renames an existing Hyper-V firewall rule. When creating a rule, if the Name parameter is not specified, then a random GUID is used. This cmdlet specifies a friendly and descriptive rule name. The newly specified name, using the NewName parameter, must still be unique since it identifies a single rule object on the computer.
+
+This cmdlet gets renames the Hyper-V firewall rule with the Name parameter, DisplayName parameter, or rule properties. Only one Hyper-V firewall rule can be renamed at a time.

```

Comment:
```suggestion

The **Rename-NetFirewallHyperVRule** cmdlet renames an existing Hyper-V firewall rule. When creating a rule, if the Name parameter is not specified, then a random GUID is used. This cmdlet specifies a friendly and descriptive rule name. The newly specified name, using the NewName parameter, must still be unique since it identifies a single rule object on the computer.

This cmdlet renames the Hyper-V firewall rule with the Name parameter, DisplayName parameter, or rule properties. Only one Hyper-V firewall rule can be renamed at a time.

```